### PR TITLE
 Fix #13808: Refactor TileInspector.cpp to use Window class

### LINKED ISSUE #13808: Refactor window to class: TileInspector
See #13737 for more details.

### PR DESCRIPTION
**Pull request by zuczek4793**
Refactor TileInspector.cpp to use the Window class instead of using static functions
- Replaced functions to avoid using functions that pass a pointer to a window
- renamed private variables
- removed unneeded case statements

If someone could check the usage of InvalidateWindow, as I couldn't delete windowTileInspectorInvalidateWindow and replace it with Invalidate like similar examples in the codebase did, as it caused issues with rendering the window.

### CODE REVIEW

**Comment by Broxzier** [comment]
Why did you close your [previous PR](https://github.com/OpenRCT2/OpenRCT2/pull/16865) and start a new one with the same changes? You can just continue working and pushing on the same branch, and the PR will update along with it.

**Comment by zuczek4793** [comment]
I renamed the branch from refactor-TileEditor to refactor-TileInspector, not realizing that it would close the PR

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 1]
```diff
@@ -510,1888 +466,1767 @@ static uint64_t PageDisabledWidgets[] = {
...
-    {
-        auto info = get_map_coordinates_from_pos(screenCoords, ViewportInteractionFlags);
-        clickedElement = info.Element;
-        mapCoords = info.Loc;
+        _tileSelected = true;
+        _toolMap = mapCoords;
+        windowTileInspectorTile = TileCoordsXY(mapCoords);
+        OpenRCT2::TileInspector::SetSelectedElement(clickedElement);
+        LoadTile(clickedElement);
     }
```
Comment (Broxzier) [inline src/openrct2-ui/windows/TileInspector.cpp:1210]: Any reason for removing empty lines like this? There are many more cases like this where they are removed, while they do improve readability.

**Review by Broxzier — CHANGES_REQUESTED** [review]
Haven't gone over all the changes yet, nor tested the changes in-game.

File: src/openrct2-ui/windows/TileInspector.cpp (hunk 1 above)
Comment (zuczek4793) [reply src/openrct2-ui/windows/TileInspector.cpp:1210]: It's been a while, but I think I removed them because they were unused

Comment (Broxzier) [reply src/openrct2-ui/windows/TileInspector.cpp:1210]: Yes, I mean the blank lines. VS is set up to use clang-format for formatting, perhaps with a newer version it's doing this?

**Comment by Broxzier** [comment]
I've squashed, rebased, merged, and formatted this PR with current develop to get things working again, and gave it a spin. Feature-wise everything still works, but the window itself has a few issues which I'll add as review points in a bit.

**Review by Broxzier — CHANGES_REQUESTED** [review]
Mostly works, but the window has a few visual bugs. See comment above.

**Comment by Broxzier** [comment]
Invalidating the window properly fixes both issues. I'll give this one more spin tomorrow, but otherwise I think it's ready to get merged.

**Review by Broxzier — APPROVED** [review]

**Pull request merged**